src/openqaoa-qiskit/openqaoa_qiskit/backends/qaoa_qiskit_qpu.py

```diff
@@ -145,19 +152,20 @@ def qaoa_circuit(self, params: QAOAVariationalBaseParams) -> QuantumCircuit:
                 cbit = self.creg[idx]
                 circuit_with_angles.measure(qubit, cbit)
 
-        if self.qaoa_descriptor.routed == True:
+        if self.qaoa_descriptor.routed is True:
             transpiled_circuit = transpile(
                 circuit_with_angles,
                 self.backend_qpu,
                 initial_layout=self.initial_qubit_mapping,
-                optimization_level=0,
+                optimization_level=self.qiskit_optimziation_level,
```

Comment:
Is this guaranteed to be always correct? We may end up routing twice, no?

Reply:
Yes, it's guaranteed since the routing_method is set to `None`. Ofcourse, as long as IBM does not change things in the future.